domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs

```diff
@@ -53,12 +52,25 @@ public async Task<CreateContractResult> Create(string gsrn, UserDescriptor user,
             return new ContractAlreadyExists(overlappingContract);
         }
 
+        var wallets = await walletClient.GetWallets(user.Subject.ToString(), cancellationToken);
+
+        var walletId = wallets.Result.FirstOrDefault()?.Id;
+        if (walletId == null)
+        {
+            var createWalletResponse = await walletClient.CreateWallet(user.Subject.ToString(), cancellationToken);
+
+            if (createWalletResponse == null)
+                throw new ApplicationException("Failed to create wallet.");
```

Comment:
Kan vi ikke give mere detaljerede fejl?

For jeg forestiller mig at en Wallet sagtens kan modtage og behandle requesten, men fejler i at give en fejl besked. I det tilfælde er fejlbeskeder misledende